targets/csharp/source/source/Uunit/PluginManagerTest.cs

```diff
@@ -0,0 +1,353 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using System.Collections.Concurrent;
+
+namespace PlayFab.UUnit
+{
+    public class PluginManagerTest : UUnitTestCase
+    {
+        private class CustomSerializerPlugin : ISerializerPlugin
+        {
+            public bool IsSerializeCalled = false;
+
+            public T DeserializeObject<T>(string serialized)
+            {
+                throw new NotImplementedException();
+            }
+
+            public T DeserializeObject<T>(string serialized, object serializerStrategy)
+            {
+                throw new NotImplementedException();
+            }
+
+            public object DeserializeObject(string serialized)
+            {
+                throw new NotImplementedException();
+            }
+
+            public string SerializeObject(object obj)
+            {
+                this.IsSerializeCalled = true;
+                return "{}";
+            }
+
+            public string SerializeObject(object obj, object serializerStrategy)
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        private class CustomTransportPlugin : ITransportPlugin
+        {
+            public string Name;
+            public bool IsDoPostCalled = false;
+
+            public async Task<object> DoPost(string urlPath, object request, Dictionary<string, string> headers)
+            {
+                this.IsDoPostCalled = true;
+                return "{}";
+            }
+        }
+
+        private class CustomApi
+        {
+            public string PluginName;
+
+            public async Task<object> SomeApiMethod()
+            {
+                var transport = (ITransportPlugin)PluginManager.GetPlugin(PluginContract.PlayFab_Transport, this.PluginName);
+                return await transport.DoPost(null, null, null);
+            }
+        };
+
+        // Functional
+        private static bool TITLE_INFO_SET = false;
+
+        // Fixed values provided from testInputs
+        private static string USER_EMAIL;
+        private static Dictionary<string, string> extraHeaders;
+
+        /// <summary>
+        /// PlayFab Title cannot be created from SDK tests, so you must provide your titleId to run unit tests.
+        /// (Also, we don't want lots of excess unused titles)
+        /// </summary>
+        public static void SetTitleInfo(TestTitleData testInputs)
+        {
+            TITLE_INFO_SET = true;
+
+            PlayFabSettings.TitleId = testInputs.titleId;
+            USER_EMAIL = testInputs.userEmail;
+            extraHeaders = testInputs.extraHeaders;
+
+            // Verify all the inputs won't cause crashes in the tests
+            TITLE_INFO_SET &= !string.IsNullOrEmpty(PlayFabSettings.TitleId)
+                && !string.IsNullOrEmpty(USER_EMAIL);
+        }
+
+        public override void SetUp(UUnitTestContext testContext)
+        {
+            if (!TITLE_INFO_SET)
```

Comment:
You shouldn't need this function-body, and once the body is empty you may not need to implement SetUp().